### DISCUSSION TITLE
How to hide Panel if I click outside of it?

### DISCUSSION THREAD

**Discussion by esenkus** [AvaloniaUI/Avalonia#9445] [General]
Hey, I tried googling a lot but could not find a solution to this, so I will be asking my question here.
What I want to do:
I want to create an button that when clicked shows a new panel (controlled by `MyPanel.IsVisible = true`) which has `ListBox` inside of it, for selecting specific keys I want. I also want that panel to be closed when I click anywhere outside of it. Basically I want what `ComboBox` does but with multiple selections.  How could I accomplish this? I tried looking into focus events, but they ar constantly changing when selecting keys in `ListBox`.

Here is a rough code of what I want:
```
...
<Panel>
<Button Name="MyButton" Click="ShowMyPanel">
            Click me
</Button>
                <!-- visible controlled from button -->
        <Panel IsVisible="False" ZIndex="1" Name="MyPanel">
                <!-- Items for now are a simple `List<string>` that I want to keep track of
            <ListBox Name="MyListBox" SelectionMode="Multiple, Toggle" Items="{Binding Items}"/>
        </Panel>
</Panel>
```

How can I force to hide this whole panel as soon as I click outside of it?

**Comment by esenkus** [comment]
I did try adding tapped event to my window
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="using:log_me_more.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        ...
        Tapped="WindowClicked">
```
and then handling it from code like:
```
private void WindowClicked(object? sender, RoutedEventArgs e) {
        if (e.Source!.InteractiveParent!.GetType() != typeof(ListBoxItem)) {
           MyPanel.IsVisible = false;
        }
    }
```
But this does not work if I click right into `TextBox` or on window bar

**Reply by lukaszossowski** [reply]
You could go an overlay approach - create a custom control that has an invisible (transparent background, not null, because null is not clickable) border behind it and set its click event to hide the whole control. Oe place the invisible border directly in your window so that you can have controls that are over and under it.

**Reply by esenkus** [reply]
That is actually a very simple and effective solution,  thanks!
